services/src/main/java/org/fao/geonet/api/pages/PageProperties.java

```diff
@@ -114,4 +123,12 @@ public String getIcon() {
     public void setIcon(String icon) {
         this.icon = icon;
     }
+
+    public List<String> getGroups() {
```

Comment:
The main concern is about not using referential integrity with the groups table. 

Could be maybe check as a similar example  the `User` -> `Address` relationship:

- https://github.com/geonetwork/core-geonetwork/blob/88cc42593123d9f50e11ae30cc87c38c6ad171bd/domain/src/main/java/org/fao/geonet/domain/User.java#L223-L244

Reply:
The challenge is the group is from JSON representation. We cannot just build the backend in pure JPA representation. The JSON representation will need to expand to future security representation not limited to groups but something else.

Reply:
@wangf1122  - @josegar74 is suggesting that the json expression be replaced with a new pageGroup table that would be used to link the pages to the groups within the database.  This will ensure that if a group or page is removed then the database will cascade the delete causing it to clean itself.  

Otherwise with the json expression, there could be issues where it references groups that no longer exist.

Reply:
@josegar74 @ianwallen 

I have tried this from my development localhost. I couldn't make it work. I guess it's due to the nature of how these tables are design. We are attempting to put the group into the page as children. So whenever page get the children remove/updated, the group will get remove/updated as well. This doesn't sounds like the right scenario to apply. It's not like the address is depended on user. If the user gets removed, address gets removed as well. 

So I will leave the current JSON design instead of building such joint table and it will mess up the group table which JPA consider all children will get wiped out if parent entity does not want it.

Once we have concrete idea. We can implement this part later. Also for the accessexpression, it servers similarly as configuration. Its very expandable in the future other than serving group constraint purpose.

Reply:
@josegar74 @ianwallen 

I have added the page/group joint table and the entity mapping. I got rid of the accessexpression JSON in this case. Please retest from your local.